### ISSUE TITLE
Allow customizations?

### ISSUE THREAD

**Issue by jonbarthol** [IdentityManager/IdentityManager.MembershipReboot#17]
We added a UserEntitlements table which is a child of the UserAccounts table. A user can have many entitlements, which for us, are digital products the user has access too. The UserEntitlements table contains various info about the digital product, such as when access starts and expires.

Adding the UserEntitlements table was super simple. We just added a virtual ICollection property to our User class and EF took care of the rest.

Is there a way to customize IdentityManager.MembershipReboot to manage custom tables like this? If not, would you accept a pull request if we were able to make it work (in a generic way)? Basically, we need a fourth "Entitlements" tab page in the UI (in addition to the Properties, Roles, and Claims tab pages).

Thanks.

**Comment by brockallen** [comment]
That's pretty much beyond the scope of IdMgr. IdMgr is about identity data not permissions.

**Comment by jonbarthol** [comment]
I'll pushback just a little, then let it go :).  ["IdMgr is about identity data not permissions."] Aren't user roles (as claims) an example of permissions? Roles are managed in IdMgr as a separate tab. And they are stored in the UserClaims table, which is a child of the UserAccounts table. We added our own UserEntitlements table, which is a sibling to UserClaims, so it would seem logical to see a new tab page in IdMgr for that new table.  Also, in your CustomUser.cs, the user has custom properties, namely First Name, Last Name, and Age. Those show up on the Properties tab in IdMgr through reflection. It would be nice to make any ICollection property show up as a new tab page.

In summary, I would say that any child, grandchild, etc of the UserAccounts table would be considered "identity data" and naturally fit into IdMgr. Thanks again.

**Comment by brockallen** [comment]
Nope, roles are identity data. Those are inputs into authorization logic, sure, but they're different (IMO).

In any event, it's custom to something you've done, so I don't see it making sense in the general IdMgr layer. If you can come up with a general approach for child data, then perhaps... but then what happens when those collections in the data model bleed into business/application data (since people do setup FKs to the user data). 

Anyway, not trying to argue... it's just that this project could be an infinite size thing and I don't really have time for it (especially doing it without funding). Feel free to fork and add your own thing -- if we see down the road it's generic enough and easy enough to add then let's chat again. Thx.

**Comment by jonbarthol** [comment]
Ok. Thanks.
